gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/oneconvergence_service_node_driver.py

```diff
@@ -0,0 +1,867 @@
+# One Convergence, Inc. and its suppliers, if any. The intellectual and
+# technical concepts contained herein are proprietary to One Convergence,
+# Inc. and its suppliers.
+#
+# Dissemination of this information or reproduction of this material is
+# strictly forbidden unless prior written permission is obtained from
+# One Convergence, Inc., USA
+
+import copy
+import yaml
+
+from keystoneclient.v2_0 import client as keyclient
+from neutron.api.rpc.agentnotifiers import dhcp_rpc_agent_api
+from neutron.api.v2 import attributes
+from neutron.common import constants as const
+from neutron.common import exceptions as n_exc
+from neutron.common import log
+from neutron import context as n_context
+from neutron.openstack.common import jsonutils
+from neutron.openstack.common import log as logging
+from neutron.plugins.common import constants as pconst
+from neutron.services.oc_service_manager.oc_service_manager_client import (
+                                                        SvcManagerClientApi)
+from oslo.config import cfg
+
+from gbpservice.neutron.services.grouppolicy.common import constants
+from gbpservice.neutron.services.servicechain.plugins.ncp import (
+                                                    exceptions as exc)
+from gbpservice.neutron.services.servicechain.plugins.ncp import model
+from gbpservice.neutron.services.servicechain.plugins.ncp.node_drivers import (
+                                openstack_heat_api_client as heat_api_client)
+from gbpservice.neutron.services.servicechain.plugins.ncp.node_drivers import (
+                                template_node_driver as template_node_driver)
+
+
+oneconvergence_driver_opts = [
+    cfg.StrOpt('svc_management_ptg_name',
+               default='svc_management_ptg',
+               help=_("Name of the PTG that is associated with the "
+                      "service management network")),
+]
+
+cfg.CONF.register_opts(oneconvergence_driver_opts, "servicechain")
+
+SC_METADATA = '{"sc_instance":"%s", "insert_type":"%s", "floating_ip": "%s", "provider_interface_mac": "%s"}'
+SVC_MGMT_PTG_NAME = cfg.CONF.servicechain.svc_management_ptg_name
+STITCHING_PTG_NAME = "traffic-stitching-gbp-internal"
+
+POOL_MEMBER_PARAMETER_AWS = {"Description": "Pool Member IP Address",
+                             "Type": "String"}
+POOL_MEMBER_PARAMETER = {"description": "Pool Member IP Address",
+                         "type": "string"}
+
+LOG = logging.getLogger(__name__)
+
+
+class InvalidServiceType(exc.NodeCompositionPluginBadRequest):
+    message = _("The OneConvergence Node driver only supports the services "
+                "VPN, Firewall and LB in a Service Chain")
+
+class ServiceInfoNotAvailableOnUpdate(n_exc.NeutronException):
+    message = _("Service information is not available with Service Manager "
+                "on node update")
+
+# REVISIT(Magesh): The Port and PT names have to be changed
+class TrafficStitchingDriver(object):
```

Comment:
Can we have a prefacing comment that the driver currently supports only L3 stitching